core/lib/common/state.js

```diff
@@ -26,7 +26,9 @@ module.exports = class State {
 			sendObject.data = this._log.join('\n');
 		}
 
-		process.send(sendObject);
+		if (process.send !== undefined) {
```

Comment:
```suggestion
		if (process.send) {
```
Would this not be cleaner and cover more conditions (like null, etc)?

Reply:
@klutchell `process.send` is undefined when the current process is not a child of another process, so we're specifically checking to see if the property is defined or not